core/capabilities/streams/mock_trigger.go

```diff
@@ -0,0 +1,470 @@
+package streams
+
+// NOTE: this file is an amalgamation of MercuryTrigger and the streams trigger load tests
+// the mercury trigger was modified to contain non-empty meta and sign the report with mock keys
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	v3 "github.com/smartcontractkit/chainlink-common/pkg/types/mercury/v3"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/chains/evmutil"
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v3/reportcodec"
+)
+
+const (
+	baseTimestamp = 1000000000
+)
+
+func RegisterMockTrigger(lggr logger.Logger, capRegistry core.CapabilitiesRegistry) (*MockTriggerService, error) {
+	ctx := context.TODO()
+	trigger := NewMockTriggerService(100, lggr)
+	if err := trigger.Start(ctx); err != nil {
+		return nil, err
+	}
+	if err := capRegistry.Add(ctx, trigger); err != nil {
+		return nil, err
+	}
+
+	producer := NewMockDataProducer(trigger, lggr)
+	if err := producer.Start(ctx); err != nil {
+		return nil, err
+	}
+
+	return trigger, nil
+}
+
+// NOTE: duplicated from trigger_test.go
+func newReport(lggr logger.Logger, feedID [32]byte, price *big.Int, timestamp int64) []byte {
+	v3Codec := reportcodec.NewReportCodec(feedID, lggr)
+	raw, err := v3Codec.BuildReport(v3.ReportFields{
+		BenchmarkPrice:     price,
+		Timestamp:          uint32(timestamp),
+		ValidFromTimestamp: uint32(timestamp),
+		Bid:                price,
+		Ask:                price,
+		LinkFee:            price,
+		NativeFee:          price,
+		ExpiresAt:          uint32(timestamp + 1000000),
+	})
+	if err != nil {
+		panic(err)
+	}
+	return raw
+}
+
+func rawReportContext(reportCtx ocrTypes.ReportContext) []byte {
+	rc := evmutil.RawReportContext(reportCtx)
+	flat := []byte{}
+	for _, r := range rc {
+		flat = append(flat, r[:]...)
+	}
+	return flat
+}
+
+const triggerID = "mock-streams-trigger@1.0.0"
```

Comment:
I like having this merged as an example trigger. I only wish it could go to common but it can't because of EVM dep :( Maybe we should create a special sub-directory with example capabilities and move on-demand trigger there as well...? @cedric-cordenier thoughts?